test/contracts/common/erc20-transfer.ts

```diff
@@ -0,0 +1,79 @@
+import {ExpectTokenTransfer} from '../bond/single-collateral-bond-events'
+import {ActualTokenTransfer} from '../bond/verify-single-collateral-bond-events'
+import {BigNumber, Event} from 'ethers'
+import {TransferEvent} from '../../../typechain-types/IERC20'
+import {expect} from 'chai'
+import {Result} from '@ethersproject/abi'
+
+export function deepEqualsERC20TokenTransfer(
+    actual: ActualTokenTransfer,
+    expected: ExpectTokenTransfer
+): boolean {
+    return (
+        actual.to === expected.to &&
+        actual.from === expected.from &&
+        actual.value.toBigInt() === expected.amount
+    )
+}
+
+/**
+ * Shape check and conversion for a TransferEvents.
+ */
+export function erc20TransferEvents(events: Event[]): {
+    from: string
+    to: string
+    value: BigNumber
+}[] {
+    const converted: {
+        from: string
+        to: string
+        value: BigNumber
+    }[] = []
+
+    for (let i = 0; i < events.length; i++) {
+        const transfer = events[i] as TransferEvent
+        expect(transfer.args).is.not.undefined
+
+        const args = events[i].args
+        expect(args?.from).is.not.undefined
+        expect(args?.to).is.not.undefined
+        expect(args?.value).is.not.undefined
+
+        converted.push(transfer.args)
+    }
+
+    return converted
+}
+export type ExpectedERC20Transfer = {
+    to: string
+    from: string
+    amount: bigint
+}
+
+export function erc20TransferEventLogsFromResult(
+    events: Result
+): ExpectedERC20Transfer[] {
+    const results: ExpectedERC20Transfer[] = []
+
+    /* eslint-disable @typescript-eslint/no-unsafe-member-access */
```

Comment:
issue: there's now code in this repo that already achieves this logic, without the need to disable linking 😉 

Converting a `Result` into a typed object:
https://github.com/windranger-io/windranger-treasury/blob/ffdbd2bc82261b70d8cc97fd0abdeccceb71d079/test/contracts/bond/bond-curator-events.ts#L32

Putting the part together for verification:
https://github.com/windranger-io/windranger-treasury/blob/ffdbd2bc82261b70d8cc97fd0abdeccceb71d079/test/contracts/bond/verify-curator-events.ts#L29

Generic ordered event verification:
https://github.com/windranger-io/windranger-treasury/blob/ffdbd2bc82261b70d8cc97fd0abdeccceb71d079/test/framework/verify.ts#L10